packages/core/src/internal/execution/types/execution-result.ts

```diff
@@ -59,6 +60,16 @@ export interface StrategyErrorExecutionResult {
   error: string;
 }
 
+/**
+ * The execution strategy returned a strategy-specific hold e.g.
+ * waiting for off-chain multi-sig confirmations.
+ */
+export interface StrategyHeldExecutionResult {
+  type: ExecutionResultType.STRATEGY_HELD;
+  heldId: number;
```

Comment:
This is different from other examples of `Results`. We infer the networkInteractionId through the network interactions. But there is no other place where the heldId exists except within the logic of the strategy.

I have opted to include the heldId in the result, so it can be fed through to the results and UI.